src/runtime/app.tsx

```diff
@@ -40,7 +41,11 @@ export function App({ helmetContext }: { helmetContext?: object }) {
     async function refetchData() {
       try {
         const pageData = await waitForApp(pathname);
+        const saved = localStorage.getItem(`${APPEARANCE_KEY}`);
         setPageData(pageData);
+        if (saved === 'dark') {
```

Comment:
What if save === 'auto'?

Reply:
I think it has no need to set appearance because:
1. global script will always be executed in mpa mode
2. the theme will reserve after navigating to other route in spa mode.